File: leetcode/src/main/ruby/reverse.rb
# @param {Integer} x
# @return {Integer}
def reverse(x)
    r, flag = 0, 1
    x, flag = -x, -1 if x < 0
    r, x = (10 * r + x % 10), x / 10 while x != 0
    r > 0x7fffffff ? 0 : r * flag
end

File: leetcode/src/main/golang/reverse.go
func reverse(x int) int {
    var r int64 = 0
    for lx := int64(x); lx != 0; lx /= 10 {
        r = 10 * r + lx % 10
    }

    if r > 0x7fffffff || r < -0x80000000 {
        return 0
    }
    return int(r)
}